src/h3lib/lib/vertex.c

```diff
@@ -269,3 +269,26 @@ void H3_EXPORT(vertexToPoint)(H3Index vertex, GeoCoord* coord) {
     // Copy from boundary to output coord
     *coord = gb.verts[0];
 }
+
+/**
+ * Whether the input is a valid H3 vertex
+ * @param  vertex H3 index possibly describing a vertex
+ * @return        Whether the input is valid
+ */
+int H3_EXPORT(isValidVertex)(H3Index vertex) {
+    if (H3_GET_MODE(vertex) != H3_VERTEX_MODE) {
+        return 0;
+    }
+
+    int vertexNum = H3_GET_RESERVED_BITS(vertex);
+    H3Index owner = vertex;
```

Comment:
Do we need the extra variable `owner` here? Could you just work with `vertex` and avoid the copy? If just for naming purposes, maybe change the input parameter name to `index` (which is maybe actually more correct, since it is perfectly valid to input a non-vertex index into this function)?

Reply:
That's a good question - I guess it should be fine to modify `vertex`, though I generally think it's confusing to modify arguments in-place. Will update.

Reply:
I'm actually going to recommend the opposite tack. Getting the `owner` H3Index is *vital* because I think this function should confirm that it can reproduce the original `vertex` value in here.

That will make this function slower, but since you'd only be validating the vertex if you're not sure about it, and since there are non-canonical representations that could be reached through some bit manipulation, it feels like the only time you'd be calling `isVertexValid` would be when you *aren't* getting them generated by the H3 library itself.

Reply:
Oh huh. I hadn't even thought of it, but yeah, the best way to check whether we have the canonical representation is to run this through `cellToVertex` again and compare output. Easier than what I was thinking, where I'd have to get neighbors again. Good idea.